### PR TITLE
[WIP] fix(cdk): fix autofocus behavior in ios

### LINKED ISSUE #1634: 🐞 - ios autofocus in dialog
### Which @taiga-ui/* package(s) are the source of the bug?

cdk

### Please provide a link to a minimal reproduction of the bug

_No response_

### Description

Dialog "jumps" when tuiAutoFocus is set on input on ios

https://user-images.githubusercontent.com/50827860/163393258-bd7eae0c-fb58-4b75-afbe-84810fa9c506.mov

s

### Angular version

11.2.11

### Taiga UI version

2.41.1

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [X] Safari
- [ ] Edge

### Which operating systems have you used?

- [ ] macOS
- [ ] Windows
- [ ] Linux
- [X] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #1634

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/cdk/directives/auto-focus/autofocus.directive.ts [hunk 1]
```diff
@@ -84,12 +86,33 @@ export class TuiAutoFocusDirective implements AfterViewInit {
         return this.element.matches('input, textarea');
     }
 
+    private isFullScreenDialog() {
```
Comment (waterplea) [inline projects/cdk/directives/auto-focus/autofocus.directive.ts:89]: This only takes into account default built-in implementation of dialogs and relies on its "private" implementation of `data-size` attribute. That's not a reliable solution and it doesn't cover all use cases.

File: projects/cdk/directives/auto-focus/autofocus.directive.ts [hunk 2]
```diff
@@ -84,12 +86,33 @@ export class TuiAutoFocusDirective implements AfterViewInit {
...
+         * @note: emulate original container size
+         */
+        fakeInput.style.height = px(this.element.clientHeight);
+        fakeInput.style.width = px(this.element.clientWidth);
+
+        /**
+         * @note:
+         * Move the focus point where the focus will browser navigate when the virtual keyboard opened
+         */
+        const offsetY = this.isFullScreenDialog() ? 0 : VIRTUAL_KEYBOARD_HEIGHT;
```
Comment (waterplea) [inline projects/cdk/directives/auto-focus/autofocus.directive.ts:113]: Doesn't it change from type to type? Even if it doesn't, this is a hardcode for a particular version of iOS virtual keyboard which can change.

File: projects/core/components/dialog/dialog.style.less [hunk 3]
```diff
@@ -53,8 +53,10 @@
     &[data-size='fullscreen'],
     &[data-size='page'] {
         width: 100vw;
+        height: 100%;
         min-height: 100%;
         border-radius: 0;
+        overflow: hidden;
```
Comment (waterplea) [inline projects/core/components/dialog/dialog.style.less:59]: Why these changes?

**Review by waterplea — COMMENTED** [review]
This solution seems really dirty to me, there has to be a better way 🤔

File: projects/core/components/dialog/dialog.style.less (hunk 3 above)
Comment (splincode) [reply projects/core/components/dialog/dialog.style.less]: fixed

File: projects/cdk/directives/auto-focus/autofocus.directive.ts (hunk 2 above)
Comment (splincode) [reply projects/cdk/directives/auto-focus/autofocus.directive.ts]: removed it

File: projects/cdk/directives/auto-focus/autofocus.directive.ts (hunk 1 above)
Comment (splincode) [reply projects/cdk/directives/auto-focus/autofocus.directive.ts]: removed it

File: projects/cdk/directives/auto-focus/autofocus.directive.ts [hunk 4]
```diff
@@ -85,27 +88,21 @@ export class TuiAutoFocusDirective implements AfterViewInit {
...
-        const fakeInput: HTMLElement = this.renderer.createElement('input');
+        const fakeInput: HTMLElement = this.createFakeInputIos();
+        const duration = this.getMinAnimationTimeBeforeFocus();
 
-        fakeInput.style.position = 'absolute';
-        fakeInput.style.opacity = '0';
-        fakeInput.style.height = '0';
-
-        const blurHandler = () => setNativeFocused(fakeInput);
+        const blurHandler = () => setNativeFocused(fakeInput, true, true);
```
Comment (nsbarsukov) [inline projects/cdk/directives/auto-focus/autofocus.directive.ts:94]: **nit**

let' make
```
const fakeInput: HTMLInputElement
```

and replace  
```
setNativeFocused(fakeInput, true, true)
```
by 
```
fakeInput.focus({preventScroll: true});
```

I think it is more readable than arguments `true, true`

Comment (splincode) [reply projects/cdk/directives/auto-focus/autofocus.directive.ts]: fixed

File: angular.json [hunk 5]
```diff
@@ -377,27 +377,20 @@
                     }
                 },
                 "test": {
-                    "builder": "@nrwl/workspace:run-commands",
```
Comment (splincode) [inline angular.json:380]: this change is needed to be able to run tests in IDEA

**Review by nsbarsukov — APPROVED** [review]

**Comment by splincode** [comment]
@waterplea what do you think about this PR?)
